Handle Azure style Group member remove operations

kjthorpe18 opened the pull request:
Azure sends PATCH operations to remove members from a group of the following form:
```
{
   "schemas":[
      "urn:ietf:params:scim:api:messages:2.0:PatchOp"
   ],
   "Operations":[
      {
         "op":"remove",
         "path":"members",
         "value":[
            {
               "value":"aaebb169-b91d-481c-9957-097faeaf649a"
            }
         ]
      }
   ]
}
```
This is not how the SCIM spec says how to remove a single member from a group, but should be easy enough to handle this case of request.

This PR adds a special case in the `DefaultPatchHandler` to convert these requests to the acceptable and spec-compliant form.

kjthorpe18 committed:
Handle Azure style Group member remove operations

bdemers commented on scim-core/src/test/java/org/apache/directory/scim/core/repository/PatchHandlerTest.java:
```diff
@@ -465,6 +467,39 @@ public void deleteItemWithComplexFilter() throws FilterParseException {
+    ));
+
+    ScimGroup updatedGroup = patchHandler.apply(group(), List.of(op));
+    assertThat(updatedGroup.getMembers().size()).isEqualTo(3);
```
nit: may want to assert on the members of the to ensure member collection contains the expected new value

bdemers commented on scim-core/src/test/java/org/apache/directory/scim/core/repository/PatchHandlerTest.java:
```diff
@@ -465,6 +467,39 @@ public void deleteItemWithComplexFilter() throws FilterParseException {
+    op.setValue(List.of(Map.of("value", "1234")));
+
+    ScimGroup updatedGroup = patchHandler.apply(group(), List.of(op));
+    assertThat(updatedGroup.getMembers()).isNotNull();
+    assertThat(updatedGroup.getMembers().size()).isEqualTo(1);
```
As an aside, (and not part of this PR, selfishly asking for #435), do you think creating custom AssertJ assertions would be helpful? or do you think it's more readable to have multiple assertions?

Maybe something, that would read like:

```java
assertThat(updatedGroup)
    .hasMemberCount(3)
    .containsMember(<expected>)
```

Mostly thinking out loud, we have a util method in PatchGeneratorTest, but I've been thinking the messages would be more clear with custom assertions.
https://github.com/apache/directory-scimple/blob/ebb01f4d022fda36f9d28315ad0ac1d9427f9ac2/scim-core/src/test/java/org/apache/directory/scim/core/repository/PatchGeneratorTest.java#L754-L762

(I've been using Hamcrest for years, so its taken longer than it should have to get comfortable with AssertJ 🤣 )

kjthorpe18 replied:
I've also really only used Hamcrest and JUnit assertions - this would be a good thing to add in a separate PR

bdemers committed:
Adds support for Azure non-standard patch REMOVE requests

Detects Azure Quirk mode.
Azure uses an out of spec patch operation that does not use an express,
but instead uses a remove with a value.  Detect this and convert it to an expression

 {
  "op":"remove",
  "path":"members",
  "value":[{
    "value":"<id>"
  }]
}

bdemers committed:
Fix: merge conflicts

bdemers commented:
@kjthorpe18 Can you rebase one more time?
After that 4ded7f3a971dc8b9441d1dc41bc1808bb36a49ac should apply clean, and we should be able to ship it!
